game-core/src/main/java/games/strategy/triplea/attachments/TerritoryEffectAttachment.java

```diff
@@ -98,6 +101,36 @@ public int getCombatEffect(final UnitType type, final boolean defending) {
     return defending ? combatDefenseEffect.getInt(type) : combatOffenseEffect.getInt(type);
   }
 
+  private void setMovementCostModifier(final String value) throws GameParseException {
+    final String[] s = splitOnColon(value);
+    if (s.length < 2) {
+      throw new GameParseException(
+          "movementCostModifier must have a count and at least one unitType" + thisErrorMsg());
+    }
+    final Iterator<String> iter = Arrays.asList(s).iterator();
+    final BigDecimal effect = getBigDecimal(iter.next());
+    while (iter.hasNext()) {
+      final String unitTypeToProduce = iter.next();
+      final UnitType ut = getData().getUnitTypeList().getUnitType(unitTypeToProduce);
+      if (ut == null) {
+        throw new GameParseException("No unit called: " + unitTypeToProduce + thisErrorMsg());
```

Comment:
Printing out the unit type list could make this exception message more helpful. For example if there is a small typo and it's not easier to tell what it is, having the list of valid values could make it more clear where/what went wrong.

Reply:
There are some maps with way too many unit types to display them all. For example, I think TWW has probably about 100.